javalib/src/main/scala/java/util/BitSet.scala

```diff
@@ -0,0 +1,859 @@
+      }
+    }
+
+    true
+    // scalastyle:on return
+  }
+
+  def intersects(set: BitSet): Boolean = {
+    // scalastyle:off return
+    val bsBits = set.bits
+    val length1 = actualArrayLength
+    val length2 = set.actualArrayLength
+
+    if (length1 <= length2) {
+      var i: Int = 0
+      while (i < length1) {
+        if ((bits(i) & bsBits(i)) != 0)
+          return true
+
+        i += 1
+      }
+    } else {
+      var i: Int = 0
+      while (i < length2) {
+        if ((bits(i) & bsBits(i)) != 0)
+          return true
+
+        i += 1
+      }
+    }
+
+    false
+    // scalastyle:on return
+  }
+
+  def cardinality(): Int = {
+    var count = 0
+
+    if (needClear) {
+      val length = bits.length
+
+      for (idx <- 0 until length) {
+        count += bitCount(bits(idx))
+      }
+    }
+
+    count
+  }
+
+  def and(set: BitSet): Unit = {
+    if (needClear) {
+      val bsBits = set.bits
+      val length1 = actualArrayLength
+      val length2 = set.actualArrayLength
+
+      if (length1 <= length2) {
+        for (i <- 0 until length1)
+          bits(i) &= bsBits(i)
+      } else {
+        for (i <- 0 until length2)
+          bits(i) &= bsBits(i)
+
+        for (i <- length2 until length1)
+          bits(i) = 0
+
+        actualArrayLength = length2
+      }
+
+      isLengthActual = !((actualArrayLength > 0) && (bits(actualArrayLength - 1) == 0))
+    }
+  }
+
+  def or(set: BitSet): Unit = {
+    val bsActualLen = set.getActualArrayLength()
+
+    if (bsActualLen > bits.length) {
+      val tempBits = new Array[Int](bsActualLen)
+      System.arraycopy(set.bits, 0, tempBits, 0, set.actualArrayLength)
+
+      for (i <- 0 until actualArrayLength)
+        tempBits(i) |= bits(i)
+
+      bits = tempBits
+      actualArrayLength = bsActualLen
+      isLengthActual = true
+    } else {
+      val bsBits = set.bits
+
+      for (i <- 0 until bsActualLen)
+        bits(i) |= bsBits(i)
+
+      if (bsActualLen > actualArrayLength) {
+        actualArrayLength = bsActualLen
+        isLengthActual = true
+      }
+    }
+
+    needClear = true
+  }
+
+  def xor(set: BitSet): Unit = {
+    val bsActualLen = set.getActualArrayLength()
+
+    if (bsActualLen > bits.length) {
+      val tempBits = new Array[Int](bsActualLen)
+      System.arraycopy(set.bits, 0, tempBits, 0, set.actualArrayLength)
+
+      for (i <- 0 until actualArrayLength)
+        tempBits(i) ^= bits(i)
+
+      bits = tempBits
+      actualArrayLength = bsActualLen
+      isLengthActual = !((actualArrayLength > 0) && (bits(actualArrayLength - 1) == 0))
+    } else {
+      val bsBits = set.bits
+
+      for (i <- 0 until bsActualLen)
+        bits(i) ^= bsBits(i)
+
+      if (bsActualLen > actualArrayLength) {
+        actualArrayLength = bsActualLen
+        isLengthActual = true
+      }
+    }
+
+    needClear = true
+  }
+
+  def andNot(set: BitSet): Unit = {
+    if (needClear) {
+      val bsBits = set.bits
+      val range =
```

Comment:
That is not a very good name for an Int ^^ Consider calling it `minLength`. Also, it's basically `Math.min(actualArrayLength, set.actualArrayLength)`.